x-pack/filebeat/input/streaming/websocket.go

```diff
@@ -164,11 +228,16 @@ func (s *websocketStream) FollowStream(ctx context.Context) error {
 
 	for {
 		select {
+		// if the keep-alive is enabled, heartbeat will be automatically cancelled when the parent context is done()
 		case <-ctx.Done():
 			s.log.Debugw("context cancelled, closing websocket connection")
 			return ctx.Err()
 		// s.tokenExpiry channel will only trigger if oauth2 is enabled and the token is about to expire
 		case <-s.tokenExpiry:
+			// cancel the keep-alive routine if enabled since we need to establish a new connection instance with the refreshed token
```

Comment:
```suggestion
			// Cancel the keep-alive routine if enabled since we need to establish a new
			// connection instance with the refreshed token.
```